packages/find-packages/src/index.ts

```diff
@@ -1,4 +1,5 @@
 import { readExactImporterManifest } from '@pnpm/read-importer-manifest'
+import '@pnpm/types'
```

Comment:
what it does when you just import types like this?

Reply:
This is so that the **TypeScript** compiler knows to reference `@pnpm/types` directly when generating implicit types for `findPkgs(…)`, otherwise it causes the compiler to fail with:
```
src/index.ts:13:16 - error TS2742: The inferred type of 'findPkgs'
cannot be named without a reference to
'../../read-importer-manifest/node_modules/@pnpm/types/lib'.
This is likely not portable. A type annotation is necessary.

13 async function findPkgs (
                  ~~~~~~~~
```

Reply:
It might be better to use:
```suggestion
import { ImporterManifest } from '@pnpm/types'
```
so that the **TypeScript** compiler doesn’t generate `import('@pnpm/types')` everywhere it’s needed, but this causes **TypeScript** to complain that it’s unused.

Reply:
This&nbsp;has&nbsp;been resolved&nbsp;in&nbsp;#2089.